packages/twenty-front/src/modules/views/components/ViewGroupsVisibilityDropdownSection.tsx

```diff
@@ -0,0 +1,115 @@
+import {
+  DropResult,
+  OnDragEndResponder,
+  ResponderProvided,
+} from '@hello-pangea/dnd';
+import { useRef } from 'react';
+import { IconEye, IconEyeOff, Tag } from 'twenty-ui';
+
+import { DropdownMenuItemsContainer } from '@/ui/layout/dropdown/components/DropdownMenuItemsContainer';
+import { StyledDropdownMenuSubheader } from '@/ui/layout/dropdown/components/StyledDropdownMenuSubheader';
+import { MenuItemDraggable } from '@/ui/navigation/menu-item/components/MenuItemDraggable';
+import {
+  RecordGroupDefinition,
+  RecordGroupDefinitionType,
+} from '@/object-record/record-group/types/RecordGroupDefinition';
+import { isDefined } from '~/utils/isDefined';
+import { DraggableItem } from '@/ui/layout/draggable-list/components/DraggableItem';
+import { DraggableList } from '@/ui/layout/draggable-list/components/DraggableList';
+
+type ViewGroupsVisibilityDropdownSectionProps = {
+  viewGroups: RecordGroupDefinition[];
+  isDraggable: boolean;
+  onDragEnd?: OnDragEndResponder;
+  onVisibilityChange: (viewGroup: RecordGroupDefinition) => void;
+  title: string;
+  showSubheader: boolean;
+  showDragGrip: boolean;
+};
+
+export const ViewGroupsVisibilityDropdownSection = ({
+  viewGroups,
+  isDraggable,
+  onDragEnd,
+  onVisibilityChange,
+  title,
+  showSubheader = true,
+  showDragGrip,
+}: ViewGroupsVisibilityDropdownSectionProps) => {
+  const handleOnDrag = (result: DropResult, provided: ResponderProvided) => {
+    onDragEnd?.(result, provided);
+  };
+
+  const getIconButtons = (index: number, viewGroup: RecordGroupDefinition) => {
+    const iconButtons = [
+      {
+        Icon: viewGroup.isVisible ? IconEyeOff : IconEye,
+        onClick: () => onVisibilityChange(viewGroup),
+      },
+    ].filter(isDefined);
+
+    return iconButtons.length ? iconButtons : undefined;
+  };
+
+  const ref = useRef<HTMLDivElement>(null);
+
+  const renderItem = (
```

Comment:
Avoid functions like that, it's a code smell that indicates that this function should be a component in another file.